<launch>
    <!-- 将 Urdf 文件的内容加载到参数服务器 -->
      <param name="robot_description" command="$(find xacro)/xacro $(find urdf_by)/urdf/xacro/car.xacro" />

    <!-- 启动 gazebo -->
    <!-- <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find gazebo_by)/worlds/box_house.world" />
    </include> -->

    <!-- 在 gazebo 中显示机器人模型 -->
    <!-- <node pkg="gazebo_ros" type="spawn_model" name="model" args="-urdf -model mycar -param robot_description"  /> -->

    <node pkg="joint_state_publisher" name="joint_state_publisher" type="joint_state_publisher" />
    <node pkg="robot_state_publisher" name="robot_state_publisher" type="robot_state_publisher" />

<node pkg="rviz" type="rviz" name="rviz" args="-d $(find A_start_by)/src/config/rviz_by.rviz"/>


<!-- 运行地图服务器，并且加载设置的地图-->
<node name="map_server1" pkg="map_server" ns="robot_1" type="map_server" args="$(find A_start_by)/src/map/1.yaml"/>
<remap from="/map" to="/map1"/>

<node name="map_server2" pkg="map_server" ns="robot_2" type="map_server" args="$(find A_start_by)/src/map/2.yaml"/> 
<remap from="/map" to="/map2"/>

<node name="map_server3" pkg="map_server" ns="robot_3" type="map_server" args="$(find A_start_by)/src/map/3.yaml"/> 
<remap from="/map" to="/map3"/>

<node name="map_server4" pkg="map_server" ns="robot_4" type="map_server" args="$(find A_start_by)/src/map/4.yaml"/> 
<remap from="/map" to="/map4"/>

<node name="map_server5" pkg="map_server" ns="robot_5" type="map_server" args="$(find A_start_by)/src/map/5.yaml"/> 
<remap from="/map" to="/map5"/>

<node pkg="tf2_ros" type="static_transform_publisher" name="son1" args="0 0 0 0 0 0 /map /odom" output="screen" />




<!-- 启动arbotix-->
    <node name="arbotix" pkg="arbotix_python" type="arbotix_driver" output="screen">
        <rosparam file="$(find urdf_by)/config/control.yaml" command="load" />
        <param name="sim" value="true" />
    </node>
    <!-- <node pkg="by_djstl" type="djstl_main" name="djstl_main" output="screen"/>
     <node pkg="A_start_by" type="A_star" name="A_star" output="screen"/> -->
</launch>
